eth/tracers/api_blocktrace.go

```diff
@@ -30,20 +32,51 @@ func (api *API) GetBlockTraceByNumberOrHash(ctx context.Context, blockNrOrHash r
 	if block.NumberU64() == 0 {
 		return nil, errors.New("genesis is not traceable")
 	}
-	if config == nil {
-		config = &TraceConfig{
-			LogConfig: &vm.LogConfig{
-				EnableMemory:     false,
-				EnableReturnData: true,
-			},
+
+	env, err := api.createTraceEnv(ctx, config, nil, block)
+	if err != nil {
+		return nil, err
+	}
+
+	return env.GetBlockTrace(block)
+}
+
+func (api *API) GetBlockTraceForCallAt(ctx context.Context, args ethapi.TransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, config *TraceCallConfig) (*types.BlockTrace, error) {
+	// Try to retrieve the specified block
+	var (
+		err   error
+		block *types.Block
+	)
+	if hash, ok := blockNrOrHash.Hash(); ok {
+		block, err = api.blockByHash(ctx, hash)
+	} else if number, ok := blockNrOrHash.Number(); ok {
+		block, err = api.blockByNumber(ctx, number)
+	} else {
+		return nil, errors.New("invalid arguments; neither block nor hash specified")
+	}
+	if err != nil {
+		return nil, err
+	}
+	if block.NumberU64() == 0 {
+		return nil, errors.New("genesis is not traceable")
+	}
+
+	block = types.NewBlockWithHeader(block.Header()).WithBody([]*types.Transaction{args.ToTransaction()}, nil)
+
+	var traceConfig *TraceConfig
+	var stateOverrides *ethapi.StateOverride
+	if config != nil {
+		traceConfig = &TraceConfig{
+			LogConfig: config.LogConfig,
+			Tracer:    config.Tracer,
```

Comment:
Would this trigger `log.Warn("Tracer params is unsupported")`?

Reply:
btw, why `Tracer` params is unsupported?

Reply:
we don't want user to specify tracer here, we only want them to use `vm.NewStructLogger(env.logConfig)`

see https://github.com/scroll-tech/go-ethereum/blob/65a3a7fc238954d48cb47d83e4b676e98dc52ac8/core/trace.go#L272-L274

Reply:
Oh, that's reasonable.